RIOT-OS/RIOT
# drivers/lpsxxx: refactor lps331ap and add support for lps25hb + lps22hb

### Contribution description

This PR is a rework of the initial ST lps331ap atmospheric pressure and temperature sensor. It does several things:
- move the original driver name from lps331ap to lpsxxx
- add support for lps25hb and lps22hb sensors
- use pseudo-modules for specific names
- add a check on the device type using the WHO_AM_I register
- change the default address from `0x44` to `0x5D` (why was it `0x44` initially, I don't know). The default address depends on the SDO/SA0 pad: when connected to ground it is `0x5D`, when connected to power supply, it is `0x5C`. On ST `x-nucleo-iks01a2` (lps22hb) and `x-nucleo-iks01a1` (lps25hb) extensions, the pad is connected to ground, thus using `0x5D` by default, makes more sense (to me)
- The driver API has changed to use return codes (specific to this driver) instead of returning the measurement values. The SAUL adaption was adapted accordingly
- A default configuration for lps22hb is provided for the ST B-l475E-IOT01A board since it embeds this sensor.

For simplicity, only basic polling mode with continuous measurements is provided by the driver. It should be possible to implement the other modes: One shot, Interrupt (with data ready) or FIFO in an extend version. Maybe a good candidate for a follow-up ? (Note that it might become a bit messy because of `#ifdef`s).




### Testing procedure

This PR can be fully tested using the IoT-LAB testbed, because some boards provides the 3 types of sensors:

1. Start an experiment on IoT-LAB with boards providing the different sensor: iotlab-m3 (lps331ap), st-iotnode (b-l475e-iot01a with lps22hb), arduino-zero (lps25hb on x-nucleo-iks01a1 extension) and nrf51dk (lps22hb on x-nucleo-iks01a2 extension):
```
$ iotlab-experiment submit -n test_pr -d 120 -l 1,archi=arduino-zero:xbee+site=saclay -l 1,archi=st-iotnode:multi+site=saclay -l 1,archi=m3:at86rf231+site=saclay -l 1,archi=nrf51dk:ble+site=saclay
```
2. Build and flash the `tests/driver_lpsxxx` test application on the different nodes and verify that it works:
```
$ make BOARD=iotlab-m3 IOTLAB_NODE=auto-ssh -C tests/driver_lpsxxx flash term
$ DRIVER=lps25hb make BOARD=arduino-zero IOTLAB_NODE=auto-ssh -C tests/driver_lpsxxx flash term
$ DRIVER=lps22hb make BOARD=nrf51dk IOTLAB_NODE=auto-ssh -C tests/driver_lpsxxx flash term
$ DRIVER=lps22hb make BOARD=b-l475e-iot01a IOTLAB_NODE=auto-ssh -C tests/driver_lpsxxx flash term
```
3. To check saul integration, build and flash the `examples/default` with the right driver module loaded (not needed on iotlab-m3 and b-l475e-iot01a as lps331ap and lps22hb are automatically provided):
```
$ make BOARD=iotlab-m3 IOTLAB_NODE=auto-ssh -C examples/default flash term
$ USEMODULE=lps25hb make BOARD=arduino-zero IOTLAB_NODE=auto-ssh -C examples/default flash term
$ USEMODULE=lps22hb make BOARD=nrf51dk IOTLAB_NODE=auto-ssh -C examples/default flash term
$ make BOARD=b-l475e-iot01a IOTLAB_NODE=auto-ssh -C examples/default flash term
```




### Issues/PRs references

closes #7267 (quite an old one, isn't it?), tick an item of #7585
